indico/modules/events/registration/client/js/form/fields/ShowIfInput.jsx

```diff
@@ -0,0 +1,84 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2025 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {useForm} from 'react-final-form';
+import {useSelector} from 'react-redux';
+
+import {FinalDropdown} from 'indico/react/forms';
+import {Fieldset} from 'indico/react/forms/fields';
+import {Translate} from 'indico/react/i18n';
+
+import {getItemsForConditionalDisplay} from '../selectors';
+
+export function ShowIfInput({hasValueSelected}) {
+  const itemsForConditionalDisplay = useSelector(state => getItemsForConditionalDisplay(state));
+  const form = useForm();
+  const [showValue, setShowValue] = useState(hasValueSelected);
+  const [, setSelectedField] = useState(null);
+  let options = [];
+  const {showIfFieldId, id: thisFieldId} = form.getState().values;
+  if (showValue) {
+    const [{choices}] = itemsForConditionalDisplay.filter(({id}) => id === showIfFieldId);
+    if (!choices) {
+      options = [
+        {value: '1', text: Translate.string('Yes')},
+        {value: '0', text: Translate.string('No')},
+      ];
+    } else {
+      options = choices.map(({caption, id}) => ({value: id, text: caption}));
+    }
```

Comment:
I think this will become a monster or require refactoring later - ideally it should be up to the field definition (registry.js) to specify what/how values for conditional display are selected.

Reply:
Yep, good idea. Changed in [5b34632](https://github.com/indico/indico/pull/6678/commits/5b34632bad748a6f71d6ca0c159d470e1df4b4d2).

Reply:
Thanks! I think now this could even be used to determine whether a particular field type can be used as a condition or not :)

Reply:
OK, I just improved (hopefully) the logic with this [352ff1e](https://github.com/indico/indico/pull/6678/commits/352ff1ec399c83be45399f257b72386fe7204972).